crates/data_components/src/s3_vectors/list_provider.rs

```diff
@@ -95,12 +108,116 @@ impl TableProvider for S3VectorsListTable {
 
     async fn scan(
```

Comment:
Should we add some unit tests to cover Partitioned Plan generation (plan snapshot)?